rust/operator-binary/src/history/history_controller.rs

```diff
@@ -440,6 +440,16 @@ fn build_stateful_set(
             ..PodSecurityContext::default()
         });
 
+    let role_group = shs
+        .rolegroup(rolegroupref)
+        .with_context(|_| CannotRetrieveRoleGroupSnafu)?;
+
+    let merged_env_vars = env_vars(
+        s3_log_dir,
+        shs.role().config.clone().env_overrides,
```

Comment:
No need to clone the entire config (including gigantic podOverrides) ;)
```suggestion
        shs.role().config.env_overrides.clone(),
```